application/views/scripts/items/search-form.php

```diff
@@ -34,71 +42,86 @@
+        //[field] = 'description'
+        //[type] = 'contains'
+        //[terms] = 'foobar'
+        //etc
         foreach ($search as $i => $rows): ?>
-            <div class="search-entry">
-                <?php
-                //The POST looks like =>
-                // advanced[0] =>
-                //[field] = 'description'
-                //[type] = 'contains'
-                //[terms] = 'foobar'
-                //etc
-                echo $this->formSelect(
-                    "advanced[$i][joiner]",
-                    @$rows['joiner'],
-                    array(
-                        'title' => __("Search Joiner"),
-                        'id' => null,
-                        'class' => 'advanced-search-joiner'
-                    ),
-                    array(
-                        'and' => __('AND'),
-                        'or' => __('OR'),
-                    )
-                );
-                echo $this->formSelect(
-                    "advanced[$i][element_id]",
-                    @$rows['element_id'],
-                    array(
-                        'title' => __("Search Field"),
-                        'id' => null,
-                        'class' => 'advanced-search-element'
-                    ),
-                    get_table_options('Element', null, array(
-                        'record_types' => array('Item', 'All'),
-                        'sort' => 'orderBySet')
-                    )
-                );
-                echo $this->formSelect(
-                    "advanced[$i][type]",
-                    @$rows['type'],
-                    array(
-                        'title' => __("Search Type"),
-                        'id' => null,
-                        'class' => 'advanced-search-type'
-                    ),
-                    label_table_options(array(
-                        'contains' => __('contains'),
-                        'does not contain' => __('does not contain'),
-                        'is exactly' => __('is exactly'),
-                        'is empty' => __('is empty'),
-                        'is not empty' => __('is not empty'),
-                        'starts with' => __('starts with'),
-                        'ends with' => __('ends with'))
-                    )
-                );
-                echo $this->formText(
-                    "advanced[$i][terms]",
-                    @$rows['terms'],
-                    array(
-                        'size' => '20',
-                        'title' => __("Search Terms"),
-                        'id' => null,
-                        'class' => 'advanced-search-terms'
-                    )
-                );
-                ?>
-                <button type="button" class="remove_search" disabled="disabled" style="display: none;"><?php echo __('Remove field'); ?></button>
+            <?php $index = $i + 1; // The index must start at 1 for accessible labels. ;?>
```

Comment:
Seems odd to need to change the indexes here for this to work; all else equal I'd like to keep these zero-indexed.

Reply:
I see now that actually this doesn't touch $i and instead makes a new $index.... this is just so that it will read "Row 1" to start and not "Row 0"? Might simply prefer to have the relevant code use $i as-is and just add 1 to $i when you're printing the `Row %s` aria-label.

Looks to me that all the search-row-$index id and labelledby stuff would work fine zero-indexed, so just using $i.

Reply:
Ready for another look.